lib/cadet_web/router.ex

```diff
@@ -26,6 +26,7 @@ defmodule CadetWeb.Router do
 
   scope "/", CadetWeb do
     get("/.well-known/jwks.json", JWKSController, :index)
+    post("/chat", ChatController, :chat)
```

Comment:
We should put this under a route that is protected with authentication; in other words, people need to log in to Source Academy @ NUS before being able to make a request to this route. Otherwise, anyone can (ab)use this API to talk to GPT-4.

Reply:
Noted with thanks.
Shifted the route to under v2.